src/flux/annotated_csv.cr

```diff
@@ -21,30 +21,28 @@ class Flux::AnnotatedCSV < CSV
 
   def initialize(string_or_io : String | IO, headers = false, @strip = false, separator : Char = DEFAULT_SEPARATOR, quote_char : Char = DEFAULT_QUOTE_CHAR)
     @parser = Parser.new(string_or_io, separator, quote_char)
-
-    loop do
-      peeked_char = @parser.peek
-      puts "peeking: #{peeked_char}"
-      break unless peeked_char == ANNOTATION_CHAR
-      cols = @parser.next_row
-      puts "found cols: #{cols}"
-      break unless cols
-
+    cols = @parser.next_row || ([] of String)
+    count = 0
+    while cols[0][0]? == ANNOTATION_CHAR
```

Comment:
I think this will raise an index missing error for the first case that `@parser.next_row` is nil

Reply:
yes, you are right, will resolve